agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -344,3 +356,345 @@ func TestTwoTasksSharedLocalVolume(t *testing.T) {
+	defer agent.Cleanup()
+
+	agent.RequireVersion(">=1.33.0")
+
+	// Setup test gmsa file
+	envProgramData := "ProgramData"
+	dockerCredentialSpecDataDir := "docker/credentialspecs"
+	programDataDir := os.Getenv(envProgramData)
+	if programDataDir == "" {
+		programDataDir = "C:/ProgramData"
+	}
+	testFileName := "test-gmsa.json"
+	testCredSpecFilePath := filepath.Join(programDataDir, dockerCredentialSpecDataDir, testFileName)
+	testCredSpecData := []byte(`{
+    "CmsPlugins":  [
+                       "ActiveDirectory"
+                   ],
+    "DomainJoinConfig":  {
+                             "Sid":  "S-1-5-21-975084816-3050680612-2826754290",
+                             "MachineAccountName":  "gmsa-acct-test",
+                             "Guid":  "92a07e28-bd9f-4bf3-b1f7-0894815a5257",
+                             "DnsTreeName":  "gmsa.test.com",
+                             "DnsName":  "gmsa.test.com",
+                             "NetBiosName":  "gmsa"
+                         },
+    "ActiveDirectoryConfig":  {
+                                  "GroupManagedServiceAccounts":  [
+                                                                      {
+                                                                          "Name":  "gmsa-acct-test",
+                                                                          "Scope":  "gmsa.test.com"
+                                                                      }
+                                                                  ]
+                              }
+}`)
+
+	err := ioutil.WriteFile(testCredSpecFilePath, testCredSpecData, 0755)
+	require.NoError(t, err)
+
+	tdOverrides := make(map[string]string)
+	tdOverrides["$$$TEST_REGION$$$"] = aws.StringValue(ECS.Config.Region)
+
+	testTask, err := agent.StartTaskWithTaskDefinitionOverrides(t, "gmsa-localfile", tdOverrides)
+	require.NoError(t, err)
+
+	// Wait for the container to start
+	testTask.WaitRunning(waitTaskStateChangeDuration)
+	taskID, err := GetTaskID(aws.StringValue(testTask.TaskArn))
+	require.NoError(t, err)
+	assert.NotEmpty(t, taskID)
+
+	// Verify container metadata
+	cid, err := agent.ResolveTaskDockerID(testTask, "windows_sample_app")
+	require.NoError(t, err, "Error resolving docker id for container in task")
+	assert.NotEmpty(t, cid)
+
+	// Setup docker client to validate credentialspec
+	endpoint := utils.DefaultIfBlank(os.Getenv(dockerEndpointEnvVariable), dockerEndpoint)
+	client, _ := sdkClient.NewClientWithOpts(sdkClient.WithHost(endpoint), sdkClient.WithVersion(sdkclientfactory.GetDefaultVersion().String()))
+
+	testCredentialSpecOption := "credentialspec=file://test-gmsa.json"
+	err = verifyContainerCredentialSpec(client, cid, testCredentialSpecOption)
+	assert.NoError(t, err)
+
+	// Stop test task
+	testTask.Stop()
+
+	err = testTask.WaitStopped(waitTaskStateChangeDuration)
+	assert.NoError(t, err)
+
+	// Cleanup the test file
+	err = os.Remove(testCredSpecFilePath)
```

Comment:
Why do you need to explicitly cleanup the test file? Isn't it TaskCleanup takes care of it? Can you add this as a verification step too?

Reply:
We cleanup explicitly for cases where the test execution is less than the task cleanup interval.